server/handler/merge_group.go

```diff
@@ -0,0 +1,94 @@
+// Copyright 2023 Palantir Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package handler
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/google/go-github/v50/github"
+	"github.com/palantir/go-githubapp/githubapp"
+	"github.com/palantir/policy-bot/policy/common"
+	"github.com/pkg/errors"
+	"github.com/rs/zerolog"
+)
+
+type MergeGroup struct {
+	Base
+}
+
+func (h *MergeGroup) Handles() []string { return []string{"merge_group"} }
+
+// Handle merge_group
+// https://docs.github.com/webhooks-and-events/webhooks/webhook-events-and-payloads#merge_group
+func (h *MergeGroup) Handle(ctx context.Context, eventType, devlieryID string, payload []byte) error {
+	var event github.MergeGroupEvent
+
+	if err := json.Unmarshal(payload, &event); err != nil {
+		return errors.Wrap(err, "failed to parse merge group event payload")
+	}
+
+	if event.GetAction() != "checks_requested" {
+		return nil
+	}
+
+	logger := zerolog.Ctx(ctx)
+	installationID := githubapp.GetInstallationIDFromEvent(&event)
+	client, err := h.NewInstallationClient(installationID)
+	if err != nil {
+		return err
+	}
+
+	if err != nil {
+		return errors.Wrap(err, "failed to create evaluation context")
+	}
+
+	repository := event.GetRepo().GetName()
+	owner := event.GetRepo().GetOwner().GetLogin()
+	mergeGroup := event.GetMergeGroup()
+	baseBranch := strings.TrimPrefix(mergeGroup.GetBaseRef(), "refs/heads/")
+	headSHA := mergeGroup.GetHeadSHA()
+
+	fetchedConfig := h.ConfigFetcher.ConfigForRepositoryBranch(ctx, client, owner, repository, baseBranch)
+
+	evalCtx := &EvalContext{
+		Client:    client,
+		Options:   h.PullOpts,
+		PublicURL: h.BaseConfig.PublicURL,
+		Config:    fetchedConfig,
+	}
+
+	_, err = evalCtx.ParseConfig(ctx, common.TriggerStatic)
```

Comment:
might have to make a new method for validating the policy since ParseConfig doesn't return an evaluator for static trigger, so as written this will post a status even if there's no policy file, and i don't think that's what we want

Reply:
We might not need to parse the configuration at all. I think it would be enough to only check `fetchedConfig` to determine if a policy file exists, since the main goal is to not post status checks in repositories that don't enable Policy Bot. If a PR is added to the merge queue, presumably the policy existed and was valid at the time of merge, so I think it could actually be confusing if the policy was changed to be invalid while the PR was in the queue and it led to a failed status.

Reply:
good point, sounds good to me, will make that change